lib/gems/pending/VMwareWebService/MiqVimInventory.rb

```diff
@@ -64,11 +65,17 @@ def initialize(server, username, password, cacheScope = nil)
         deleteProperty(:HostSystem, "config.dateTimeInfo")
       end
     else
-      if @v4 && cacheScope != :cache_scope_event_monitor
+      if @v4
         @propMap = dupProps(@propMap)
-        addProperty(:VirtualMachine, "runtime.memoryOverhead")
-        deleteProperty(:VirtualMachine, "config.hardware.numCoresPerSocket")
+
+        deleteProperty(:Datastore, "summary.maintenanceMode")
+
+        if cacheScope != :cache_scope_event_monitor
+          addProperty(:VirtualMachine, "runtime.memoryOverhead")
+          deleteProperty(:VirtualMachine, "config.hardware.numCoresPerSocket")
```

Comment:
Yeah I get the `addProperty` needing to be in there, but why is this [`deleteProperty(:VirtualMachine, "config.hardware.numCoresPerSocket")` in `@v2`](https://github.com/ManageIQ/manageiq-gems-pending/blob/ba8878680bb41af9ca22e17673316e0aa0ac3de9/lib/gems/pending/VMwareWebService/MiqVimInventory.rb#L55) different than this [`deleteProperty(:VirtualMachine, "config.hardware.numCoresPerSocket")` in `@v4`](https://github.com/ManageIQ/manageiq-gems-pending/blob/ba8878680bb41af9ca22e17673316e0aa0ac3de9/lib/gems/pending/VMwareWebService/MiqVimInventory.rb#L72)  one is in `if cacheScope != :cache_scope_event_monitor` but the other is not.  Maybe it was kind of an optimization since at that point nothing had to be done for `@v4` if it was an event monitor?  Just is kind of confusing and the deleteProperty should work if there is nothing in the prop map anyway...